packages/core/src/checker/checker-facade.ts

```diff
@@ -34,73 +35,67 @@ export class CheckerFacade {
     };
   }
 
-  private async executeChecker(
-    checkerIndex: number,
-    previousPassedMutants$: Observable<MutantRunPlan>,
-    checkResult$: Subject<MutantResult>,
-    passedMutant$: Subject<MutantRunPlan>
-  ) {
-    const mutants = await lastValueFrom(merge(previousPassedMutants$).pipe(toArray()));
-
-    const groups = await firstValueFrom(
-      this.checkerPool.schedule(of(0), async (checker) => {
-        const mutantArray = mutants.map((m) => m.mutant);
-        const group = await checker.createGroups?.(checkerIndex, mutantArray);
-
-        return group ?? mutants.map((m) => [m.mutant]);
+  private executeChecker(checkerName: string, input$: Observable<MutantRunPlan>) {
+    const group$ = this.checkerPool
+      .schedule(input$.pipe(bufferTime(CHECK_BUFFER_MS)), async (checker, mutants) => {
+        const groupedMutantIds = await checker.group(
+          checkerName,
+          mutants.map(({ mutant }) => mutant)
+        );
+        this.log.debug('Checker %s created %s groups (based on %s mutants).', checkerName, groupedMutantIds.length, mutants.length);
+        return groupedMutantIds.map((mutantIdGroup) => mutantIdGroup.map((mutantId) => mutants.find(({ mutant }) => mutant.id === mutantId)!));
       })
-    );
-
-    this.log.debug(`${this.options.checkers[checkerIndex]} created ${groups.length} groups.`);
-
-    const run$ = this.checkerPool.schedule(from(groups), async (checker, mutantGroup) => {
-      const results = await checker.check(checkerIndex, mutantGroup);
-      Object.entries(results).forEach(([id, result]) => {
-
-        /**
-         * TODO: It searches in the whole mutant array. It should only search in the mutant group.
-         */
-        const mutant = mutants.find((m) => m.mutant.id === id)!;
-
-        if (result.status === CheckStatus.Passed) passedMutant$.next(mutant);
-        else checkResult$.next(this.mutationTestReportHelper.reportCheckFailed(mutant.mutant, result));
-      });
-    });
-
-    await lastValueFrom(run$);
-    passedMutant$.complete();
+      .pipe(mergeMap((mutantGroups) => mutantGroups));
+    const checkTask$ = this.checkerPool
+      .schedule(group$, async (checker, group) => {
+        const checkResults = await checker.check(
+          checkerName,
+          group.map(({ mutant }) => mutant)
+        );
+        return group.map((mutantPlan) => {
+          const checkResult = checkResults[mutantPlan.mutant.id];
+          if (checkResult.status === CheckStatus.Passed) {
+            return mutantPlan;
+          } else {
+            return {
+              plan: PlanKind.EarlyResult as const,
+              mutant: mutantPlan.mutant,
+              checkResult,
+            };
+          }
+        });
+      })
+      .pipe(
+        mergeMap((mutantGroupResults) => mutantGroupResults),
+        shareReplay()
+      );
+    const [passedCheckResult$, failedCheckResult$] = partition(checkTask$, (item): item is MutantRunPlan => item.plan === PlanKind.Run);
```

Comment:
Instead of creating subjects and having to manage those instances, I use `partition` as a way to split an `observable` into 2. Using `shareReplay` will use a `ReplaySubject` under the hood to make sure the underlying observable is only executed once.